pkg/kubelet/server/server.go

```diff
@@ -878,6 +890,39 @@ func (s *Server) getPortForward(request *restful.Request, response *restful.Resp
 	proxyStream(response.ResponseWriter, request.Request, url)
 }
 
+// checkpoint handles the checkpoint API request. It checks if the requested
+// podNamespace, pod and container actually exist and only then calls out
+// to the runtime to actually checkpoint the container.
+func (s *Server) checkpoint(request *restful.Request, response *restful.Response) {
+	if request == nil || response == nil {
+		return
+	}
+	pod, ok := s.host.GetPodByName(request.PathParameter("podNamespace"), request.PathParameter("podName"))
+	if !ok {
+		response.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	containerName := request.PathParameter("containerName")
+
+	found := false
+	for _, container := range pod.Spec.Containers {
```

Comment:
It might or might not be necessary to have the checkpoint of every container if there are multiple replicas of the same container. Not sure, if it would scale well if there are 1000s of replicas of containers and corresponding checkpoints for each replica. This would further occupy space on disk if local or registry if remote. Maybe an option to just have 1 checkpoint for each replica set would be handy.  Any thoughts?